src/main/webapp/js/user-page-loader.js

```diff
@@ -76,86 +58,71 @@ function buildMessageDiv(message) {
   return messageDiv;
 }
 
-// Change the content of the page to the current page content
-const POSTS_PER_PAGE = 5;
-var currentPage = 1;
-function changePage(page) {
-  var btnForward = document.getElementById("btn_forward");
-  var btnBack = document.getElementById("btn_back");
-  const url = '/feed';
+/**
+  Fetches endpoint and retrieve messages from a specific endpoint.
+  Stores previous cursors for the back button and gets the next
+  cursors. Builds the message body.
+*/
+var cursors = {}
+// Fetch messages and add them to the page.
+function fetchMessages(currentCursor) {
+  var url = '/feed';
+  if (!currentCursor) {
+    url = '/feed';
+  } else {
+    url = '/feed?cursor=' + currentCursor;
+  }
   fetch(url).then((response) => {
-      return response.json();
-   }).then((messages) => {
-    const NUM_PAGES = Math.ceil(messages.length / POSTS_PER_PAGE);
-    var messageContainer = document.getElementById("message-container");
-
-    // Validate page
-    if (page < 1) page = 1;
-    if (page > NUM_PAGES) page = NUM_PAGES;
-
-    if(messages.length == 0){
-     messageContainer.innerHTML = '<p>No posts on this page.</p>';
+    return response.json();
+  }).then((messages) => {
+    const messageContainer = document.getElementById('message-container');
+    if (messages.length == 0) {
+      messageContainer.innerHTML = '<p>There are no posts yet.</p>';
     }
     else {
-     messageContainer.innerHTML = '';
+      messageContainer.innerHTML = '';
     }
-    for (var i = (page-1) * POSTS_PER_PAGE; i < (page * POSTS_PER_PAGE) && i < messages.length; i++) {
+    for (var i = 0; i < messages.length; i++) {
```

Comment:
Move the logic that updates the div to a separate function and call it addMessagestoDiv or something like that.

Reply:
Good point, fixed